### ISSUE TITLE
Missing WebKitGTK runs for Dec 21-22

### ISSUE THREAD

**Issue by stephenmcgruer** [web-platform-tests/wpt#20902]
From Dec 21st onwards, WebKitGTK runs are failing to complete on `epochs/daily`. Looks like wpt-webkitgtk_minibrowser-nightly-crashtest-1 always fails, with:

```
ERROR Unsupported test type crashtest for product webkitgtk_minibrowser
 0:10.31 INFO Closing logging queue
 0:10.31 INFO queue closed
 0:10.31 CRITICAL No tests ran
```
https://community-tc.services.mozilla.com/tasks/KDU0X3hGRdae96b9fEtqzQ/runs/0/logs/https%3A%2F%2Fcommunity-tc.services.mozilla.com%2Fapi%2Fqueue%2Fv1%2Ftask%2FKDU0X3hGRdae96b9fEtqzQ%2Fruns%2F0%2Fartifacts%2Fpublic%2Flogs%2Flive.log

I'm not immediately sure what change caused this to happen, so firing off a bit of a wide net: @gsnedders @jgraham @clopez @Hexcles

**Comment by clopez** [comment]
I see that a new type of test (crashtest) has been added to wptrunner, and the product runner for webkitgtk_minibrowser still doesn't support it.

However [that already happened on dec 12](https://github.com/web-platform-tests/wpt/commit/40113cc91c00f0cad78072d2fd51805a775df95c), according to the git log history.
So runs for WebKitGTK should be failing since then, unless the results for wpt.fyi are uploaded anyway this step fails. And since we have results on wpt.fyi beyond that point I guess the results are being uploaded anyways.

On the taks group for this TC run I see another step that also failed for webkitgtk, which is [wpt-webkitgtk_minibrowser-nightly-testharness-2](https://community-tc.services.mozilla.com/tasks/OHZrdHakTNCXARdhOKP20g) Looking into that failure I see:
```
  File "/home/test/web-platform-tests/tools/webdriver/webdriver/client.py", line 470, in send_command
    session=self)
  File "/home/test/web-platform-tests/tools/webdriver/webdriver/transport.py", line 175, in send
    response = self._request(method, uri, payload, headers)
  File "/home/test/web-platform-tests/tools/webdriver/webdriver/transport.py", line 191, in _request
    return self.connection.getresponse()
  File "/usr/lib/python2.7/httplib.py", line 1137, in getresponse
    response.begin()
  File "/usr/lib/python2.7/httplib.py", line 448, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python2.7/httplib.py", line 412, in _read_status
    raise BadStatusLine("No status line received - the server has closed the connection")
BadStatusLine: No status line received - the server has closed the connection
```
It looks like the WebKitWebdriver hanged or something like that.

In any case, I'm looking into adding support for running crashtest tests for webkitgtk.

**Comment by clopez** [comment]
> In any case, I'm looking into adding support for running crashtest tests for webkitgtk.

Opened PR #20914 for that

**Comment by clopez** [comment]
I see the run for WebKitGTK that was missed was the one from 22th Dec:

https://wpt.fyi/runs?label=master&max-count=100&product=webkitgtk

Which was the run for commit [ba353b09d0](https://github.com/web-platform-tests/wpt/commit/ba353b09d0) and it seems it was related with the failure for -testharness-2  commented above and not caused by the failure on -crashtest-1.

In any case, thanks for reporting this, I missed this new crashtest tests that we were completely failing to run. They should be working now. I will keep and eye on new runs to ensure things are stable.

**Comment by stephenmcgruer** [comment]
Sorry I missed that it was actually `wpt-webkitgtk_minibrowser-nightly-testharness-2` that was failing -_-. Thanks for looking into this and fixing it - feel free to close once you feel things are stable.

**Comment by clopez** [comment]
The step for crashtest its now working as expected.
However, [checking the past runs](https://wpt.fyi/runs?label=master&max-count=100&product=webkitgtk) I see that On Dec 25 the run for WebKitGTK [failed on the step -testharness-6](https://community-tc.services.mozilla.com/tasks/VCQtQ_QgRFytbyMrCfGKmg)

The failure looks similar (if not the same) than the previous one from [-testharness-2 ](https://github.com/web-platform-tests/wpt/issues/20902#issuecomment-568560375)

```
CRITICAL Traceback (most recent call last):
  File "/home/test/web-platform-tests/tools/wptrunner/wptrunner/testrunner.py", line 126, in run_test
    return self.executor.run_test(test)
  File "/home/test/web-platform-tests/tools/wptrunner/wptrunner/executors/base.py", line 254, in run_test
    self.on_environment_change(test.environment)
  File "/home/test/web-platform-tests/tools/wptrunner/wptrunner/executors/executorwebdriver.py", line 360, in on_environment_change
    self.protocol.testharness.load_runner(new_environment["protocol"])
  File "/home/test/web-platform-tests/tools/wptrunner/wptrunner/executors/executorwebdriver.py", line 93, in load_runner
    self.webdriver.url = url
  File "/home/test/web-platform-tests/tools/webdriver/webdriver/client.py", line 19, in inner
    return func(self, *args, **kwargs)
  File "/home/test/web-platform-tests/tools/webdriver/webdriver/client.py", line 528, in url
    return self.send_session_command("POST", "url", body)
  File "/home/test/web-platform-tests/tools/webdriver/webdriver/client.py", line 515, in send_session_command
    return self.send_command(method, url, body)
  File "/home/test/web-platform-tests/tools/webdriver/webdriver/client.py", line 470, in send_command
    session=self)
  File "/home/test/web-platform-tests/tools/webdriver/webdriver/transport.py", line 175, in send
    response = self._request(method, uri, payload, headers)
  File "/home/test/web-platform-tests/tools/webdriver/webdriver/transport.py", line 191, in _request
    return self.connection.getresponse()
  File "/usr/lib/python2.7/httplib.py", line 1137, in getresponse
    response.begin()
  File "/usr/lib/python2.7/httplib.py", line 448, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python2.7/httplib.py", line 412, in _read_status
    raise BadStatusLine("No status line received - the server has closed the connection")
BadStatusLine: No status line received - the server has closed the connection
```

I will try to investigate why this is happening

**Comment by clopez** [comment]
The [buildbot producing the WebKitGTK nightly bundles for testing on the WPT CI](https://build-webkit.igalia.com/builders/GTK%20Linux%2064-bit%20WPT%20Packaging%20%28nightly%20channel%29?numbuilds=50) has been failing to build since a week ago (Dec 24), so the WPT CI has been testing with WebKitGTK r253883 (the last one that was built sucessfully on Dec 23). See [WebKit bug 205649](https://bugs.webkit.org/show_bug.cgi?id=205649) for what caused this build failure.
I hope this issue with the WebDriver timeouts gets better when the CI starts running back with fresher night-lies.

**Comment by LukeZielinski** [comment]
So looks like we're getting consistent WebKitGTK runs now. @clopez is there anything more to do here?

**Comment by Hexcles** [comment]
Tentatively downgrading to roadmap

**Comment by clopez** [comment]
On Jan7 another run was missed for the same issue. Check[ this table with servo, that triggers at the same intervals](https://wpt.fyi/runs?max-count=100&product=servo&product=webkitgtk). The [issue](CCmUxlRqSzv68DdrzGJg) its the same than commented [above](https://github.com/web-platform-tests/wpt/issues/20902#issuecomment-569309845). I tried to reproduce it locally without luck so far.

**Comment by clopez** [comment]
In this issue there are two things mixed:
* Crash-test support: fixed on #20914
* The test step fails because the webdriver is not responding: Let's move that to #21594

Closing this
